Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -669,6 +670,9 @@ ppExplanation (UnknownArch unknownArches) =
     "Unknown architecture name " ++ commaSep (map quote unknownArches)
 ppExplanation (UnknownCompiler unknownImpls) =
     "Unknown compiler name " ++ commaSep (map quote unknownImpls)
+ppExplanation (MissingUpperBounds pname) =
+  "'" ++ unPackageName pname ++ "' misses upper bounds, add them"
```

Comment:
A *list* of dependencies missing upper bounds would e great (plus, the specific target).